contracts/SproutToken.sol

```diff
@@ -224,20 +224,28 @@ contract SproutToken is ERC20Burnable {
      * @return stakeBalance Staked ECO amount.
      * @return lastDeposit Last deposit timestamp in seconds since epoch.
      * @return lastMint Last mint timestamp in seconds since epoch.
+     * @return blockTime Current block time.
      * @dev The returned SPRT balance is not taking the current stake into 
      * @dev account. This is required to know the actual amount of SPRT after 
      * @dev extrapolation since `balanceOf` accounts for the current stake.
      */
     function generationExtrapolationInformation(address _account)
     public
     view
-    returns (uint256 rawBalance, uint256 stakeBalance, uint256 lastDeposit, uint256 lastMint)
+    returns (
+        uint256 rawBalance,
+        uint256 stakeBalance,
+        uint256 lastDeposit,
+        uint256 lastMint,
+        uint256 blockTime
+    )
     {
         return (
             super.balanceOf(_account),
             mStakeBalance[_account],
             mLastDeposit[_account],
-            mLastMint[_account]
+            mLastMint[_account],
+            block.timestamp
```

Comment:
Can you confirm this only useful in order not to have to do 2 calls to the node like:

```typescript
// pseudo code
const info = sproutToken.generationExtrapolationInformation();
const timestamp = provider.getBlock(provider.getBlockNumber()).timestamp;
```

If so, why ? Performance issue ? Or dev/debug mode issue ?

Reply:
> WRT @megaturbo comment i'm not sure if its better to have in `generationExtrapolationInformation` or not (it would save a bit of gas on deploy but if we're always using it, is it cleaner as is?)

I did a bit of research on the subject and I cannot say which one is the best. I think the gas is really not the issue here so we should go with the cleanest and most straightforward. However in this case there's no clear winner to me, I guess having the `block.timestamp` in the returned value ensure that no block were mined between the two calls.

I resolve this one